powersimdata/data_access/data_access.py

```diff
@@ -93,6 +93,57 @@ def close(self):
         pass
 
 
+class LocalDataAccess(DataAccess):
+    """Interface to shared data volume"""
+
+    def __init__(self, root=None):
+        self.root = root if root else server_setup.DATA_ROOT_DIR
+
+    def copy_from(self, file_name, from_dir):
+        """Copy a file from data store to userspace.
+
+        :param str file_name: file name to copy.
+        :param str from_dir: data store directory to copy file from.
+        """
+        pass
+
+    def copy_to(self, file_name, to_dir, change_name_to=None):
+        """Copy a file from userspace to data store.
+
+        :param str file_name: file name to copy.
+        :param str to_dir: data store directory to copy file to.
+        :param str change_name_to: new name for file when copied to data store.
+        """
+        self.check_filename(file_name)
+        src = posixpath.join(server_setup.LOCAL_DIR, file_name)
+        file_name = file_name if change_name_to is None else change_name_to
+        dest = posixpath.join(self.root, to_dir, file_name)
+        self.check_file_exists(dest, should_exist=False)
+        self.copy(src, dest)
+        self.remove(src)
+
+    def execute_command(self, command):
+        """Execute a command locally at the data access.
+
+        :param list command: list of str to be passed to command line.
+        """
+
+        def wrap(s):
+            if s is not None:
+                return s
+            return open(os.devnull)
+
+        proc = Popen(
+            command,
+            shell=True,
+            executable="/bin/bash",
+            stdout=PIPE,
+            stderr=PIPE,
+            text=True,
+        )
+        return wrap(None), wrap(proc.stdout), wrap(proc.stderr)
```

Comment:
Not sure I understand the need for the first element of the returned tuple. Do you redirect the standard streams to dev/null?

Reply:
The implementation in `SSHDataAccess` uses `exec_command` from paramiko which returns the standard stream tuple with 3 elements - we have to match that here. We don't ever use the stdin component so I omitted that. The `wrap` is mostly for the `stderr` part which, when successful, will be `None` and cause an error when we call `.readlines()` - compared to paramiko which gives an empty stream. To fix that we just read from `/dev/null` here.